### ISSUE TITLE
Tarantoolctl rocks install not working when tarantool is installed outside /usr

### ISSUE THREAD

**Issue by rosik** [tarantool/tarantool#3148]
I perform tarantool installation this way:

```bash
mkdir /tmp/build
cd /tmp/build
git clone https://github.com/tarantool/tarantool.git /opt/tarantool-src
cmake -DCMAKE_INSTALL_PREFIX=/opt/tarantool-install -DENABLE_DIST=ON -DCMAKE_BUILD_TYPE=Debug /opt/tarantool-src && make -j && make install
echo "export PATH=/opt/tarantool-install/bin/:$PATH" >> ~/.bashrc
source ~/.bashrc
```

Then I try to install ckit rockspec and it fails:
```bash
git clone https://github.com/tarantool/modulekit.git /opt/modulekit
cd /opt/modulekit
git checkout ckit
tarantoolctl rocks make ckit-scm-1.rockspec
```
And it says
```console
$ tarantoolctl rocks make ckit-scm-1.rockspec

Error: Could not find header file for TARANTOOL
  No file module.h in /usr/local/include
  No file module.h in /usr/include
You may have to install TARANTOOL in your system and/or pass TARANTOOL_DIR or TARANTOOL_INCDIR to the luarocks command.
Example: luarocks install ckit TARANTOOL_DIR=/usr/local
```

**Comment by rosik** [comment]
For this problem to be solved the following requirements should be fulfilled:

1. luarocks should be able to find `CMAKE_INSTALL_PREFIX=/opt/tarantool-install` through `external_deps_dirs` config parameter.
It could be done by adding it to the site_config.lua

2. rockspec should pass `TARANTOOL_DIR` to cmake variables
```
build = {
    type = 'cmake';
    variables = {
        TARANTOOL_DIR="$(TARANTOOL_DIR)";
        -- ...
    };
}
```

3. FindTarantool.cmake should properly use find_path() cmake function:
```
find_path(TARANTOOL_INCLUDE_DIR tarantool/module.h
  HINTS ${TARANTOOL_DIR} ENV TARANTOOL_DIR
  PATH_SUFFIXES include
)
```

**Comment by rosik** [comment]
I have already fixed items 2, 3 earlier in [modulekit repo](https://github.com/tarantool/modulekit/commit/676344bd0eb60cf708b36eab9e78b01af37a22bd)
Item 1 is still to be fixed in tarantool repo

**Comment by locker** [comment]
Fixed by 3171288c5d77888ba6de0e2a68b59fdd09b6afde https://github.com/tarantool/luarocks/commit/b2dde9a4cede89aa985e26aa339e4ea340199101
